File: agents/clean-code-analyzer/scripts/tests/e2e/fixtures/clean_go/clean_service.go
package main

const SecondsPerDay = 86400

type UserRepository interface {
    FindByID(userID string) (*User, error)
}

type UserService struct {
    repo UserRepository
}

func NewUserService(repo UserRepository) *UserService {
    return &UserService{repo: repo}
}

func (s *UserService) GetUser(userID string) (*User, error) {
    user, err := s.repo.FindByID(userID)
    if err != nil {
        return nil, fmt.Errorf("get user %s: %w", userID, err)
    }
    return user, nil
}

File: agents/clean-code-analyzer/scripts/tests/e2e/fixtures/clean_typescript/clean_service.ts
const SECONDS_PER_DAY = 86400;

interface UserRepository {
  findById(userId: string): Promise<User>;
}

class UserService {
  constructor(private readonly repository: UserRepository) {}

  async getUser(userId: string): Promise<User> {
    return this.repository.findById(userId);
  }
}
